compiler/modulegraphs.nim

```diff
@@ -712,12 +726,44 @@ func belongsToStdlib*(graph: ModuleGraph, sym: PSym): bool =
 proc `==`*(a, b: SymInfoPair): bool =
   result = a.sym == b.sym and a.info.exactEquals(b.info)
 
+func cmp*(a: SymInfoPair; b: SymInfoPair): int =
+  result = cmp(int32(a.info.fileIndex), int32(b.info.fileIndex))
+  if result == 0:
+    result = cmp(a.info.line, b.info.line)
+  if result == 0:
+    result = cmp(a.info.col, b.info.col)
+  if result == 0:
+    result = cmp(a.isDecl, b.isDecl)
+proc sort*(s: var SuggestFileSymbolDatabase) =
+  s.items.sort(cmp)
+  s.isSorted = true
+
+proc add*(s: var SuggestFileSymbolDatabase; v: SymInfoPair) =
+  s.items.add(v)
+  s.isSorted = false
+
+proc add*(s: var SuggestSymbolDatabase; v: SymInfoPair) =
+  s.mgetOrPut(v.info.fileIndex, newSuggestFileSymbolDatabase()).add(v)
+
+proc findSymInfo*(s: var SuggestFileSymbolDatabase; li: TLineInfo): ref SymInfoPair =
+  if not s.isSorted:
+    s.sort()
+  var q = SymInfoPair(
+    info: li
+  )
+  var idx = binarySearch(s.items, q, cmp)
+  if idx != -1:
+    new(result)
+    result[] = s.items[idx]
```

Comment:
Why not use `result = addr s.items[idx]`?

Reply:
Done: https://github.com/nim-lang/Nim/commit/6a22c4a2ca6bb672dc440b90b47acb8112b19c21